hbase-server/src/test/java/org/apache/hadoop/hbase/replication/TestMasterReplication.java

```diff
@@ -441,6 +442,41 @@ public void testCyclicReplication3() throws Exception {
     }
   }
 
+  /**
+   * Tests that base replication peer configs are applied on peer creation
+   * and the is overriden if updated as part of updateReplicationPeerConfig()
+   *
+   */
+  @Test
+  public void testBasePeerConfigsAppliedOnPeerCreationAndUpdate()
+    throws Exception {
+    LOG.info("testBasePeerConfigsAppliedOnPeerCreation");
+    String customPeerConfigKey = "hbase.xxx.custom_config";
+    String customPeerConfigValue = "test";
+    String customPeerConfigUpdatedValue = "test_updated";
+    try {
+      baseConfiguration.set(ReplicationPeerConfigUtil.HBASE_REPLICATION_PEER_BASE_CONFIG,
+        customPeerConfigKey.concat("=").concat(customPeerConfigValue));
+      startMiniClusters(2);
+      addPeer("1", 0, 1);
+      Admin admin = utilities[0].getAdmin();
+
+      Assert.assertEquals(customPeerConfigValue, admin.getReplicationPeerConfig("1").
+        getConfiguration().get(customPeerConfigKey));
+
+      ReplicationPeerConfig updatedReplicationPeerConfig = ReplicationPeerConfig.
+        newBuilder(admin.getReplicationPeerConfig("1")).
+        putConfiguration(customPeerConfigKey,customPeerConfigUpdatedValue).build();
+      admin.updateReplicationPeerConfig("1", updatedReplicationPeerConfig);
```

Comment:
Can you also add some tests for deleting a configuration? delete an override, base config should be picked up if it is present

Also, add another peer and make sure it only has the base configuration and not the updated values above.

Also, add a peer, update base configuration, restart cluster, new base should be picked up (our target usecase essentially).

Reply:
Looks like hbase does not provide the support for deleting a peer configuration, we can only update the value after a configuration is added. But I have added the test case for the scenario that new base config gets picked up after hbase restart.